Backend/Remora.Discord.Gateway/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -75,22 +76,69 @@ Func<string> token
         )
             where TResponder : IResponder
         {
-            var responderTypeInterfaces = typeof(TResponder).GetInterfaces();
+            return serviceCollection.AddResponder(typeof(TResponder));
+        }
+
+        /// <summary>
+        /// Adds all exported responders in the provided assembly to the service collection.
+        /// This method registers the responder as being available for all <see cref="IResponder{T}"/>
+        /// implementations it supports.
+        /// </summary>
+        /// <param name="serviceCollection">The service collection.</param>
+        /// <param name="assembly">The assembly to search.</param>
+        /// <returns>The service collection, with the responder added.</returns>
+        public static IServiceCollection AddRespondersFromAssembly
+        (
+            this IServiceCollection serviceCollection,
+            Assembly assembly
+        )
+        {
+            // ReSharper disable once SuspiciousTypeConversion.Global
+            var types = assembly.GetExportedTypes()
```

Comment:
This won't work, since `t` will always be an instance of `Type`, which does not implement `IResponder`. You'll need to check the interfaces implemented by the type instead.

Reply:
... I must not have been thinking when I wrote that code. I'll fix that up in a few moments.